Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverMode.hs

```diff
@@ -40,5 +41,14 @@ setDriverMode apiKey req = do
   locationTrackingServiceKey <- asks (.locationTrackingServiceKey)
   unless (apiKey == Just locationTrackingServiceKey) $ do
     throwError $ InvalidRequest "Invalid API key"
-  void $ QDriverInformation.updateActivity isActive (Just mode) driverId
+  let newFlowStatus = getDriverFlowStatus (Just mode) isActive
+  void $ QDriverInformation.updateActivity isActive (Just mode) (Just newFlowStatus) driverId
   pure Success
+
+getDriverFlowStatus :: Maybe DriverInfo.DriverMode -> Bool -> DDFS.DriverFlowStatus
+getDriverFlowStatus mode isActive =
+  case mode of
+    Just DriverInfo.ONLINE -> DDFS.ONLINE
+    Just DriverInfo.SILENT -> DDFS.SILENT
+    Just DriverInfo.OFFLINE -> DDFS.OFFLINE
+    Nothing -> if isActive then DDFS.ACTIVE else DDFS.INACTIVE
```

Comment:
ACTIVE and INACTIVE related to subscription balance, as @0utkarsh mentioned.
Maybe it should be ONLINE and OFFLINE?